aiops/opni-opensearch-update-service/opensearch-update-service/main.py

```diff
@@ -0,0 +1,148 @@
+
+
+pretrained_model_script_source = 'ctx._source.anomaly_level = ctx._source.anomaly_predicted_count != 0 ? "Anomaly" : "Normal";'
+workload_script_source = 'ctx._source.anomaly_level = ctx._source.anomaly_predicted_count == 0 ? "Normal" : ctx._source.anomaly_predicted_count == 1 ? "Suspicious" : "Anomaly";'
+pretrained_model_script_source += "ctx._source.opnilog_confidence = params['opnilog_score'];"
+workload_script_source += "ctx._source.opnilog_confidence = params['opnilog_score'];"
+script_for_anomaly = (
+    "ctx._source.anomaly_predicted_count += 1; ctx._source.opnilog_anomaly = true;"
+)
+
+
+async def doc_generator(df):
+    main_doc_keywords = {"_op_type", "_index", "_id", "doc"}
+    df["_op_type"] = "update"
+    df["_index"] = "logs"
+    df.rename(columns={"log_id": "_id"}, inplace=True)
+    df["entry_processed"] = True
+    for index, document in df.iterrows():
+        doc_dict = document.to_dict()
+        doc_dict["doc"] = {}
+        doc_dict_keys = list(doc_dict.keys())
+        for k in doc_dict_keys:
+            if not k in main_doc_keywords:
+                doc_dict["doc"][k] = doc_dict[k]
+                del doc_dict[k]
+        yield doc_dict
+
+
+
+async def setup_es_connection():
+    ES_ENDPOINT = os.environ["ES_ENDPOINT"]
+    ES_USERNAME = os.environ["ES_USERNAME"]
+    ES_PASSWORD = os.environ["ES_PASSWORD"]
+    # This function will be setting up the Opensearch connection.
+    logging.info("Setting up AsyncElasticsearch")
+    return AsyncElasticsearch(
+        [ES_ENDPOINT],
+        port=9200,
+        http_auth=(ES_USERNAME, ES_PASSWORD),
+        http_compress=True,
+        max_retries=10,
+        retry_on_status={100, 400, 503},
+        retry_on_timeout=True,
+        timeout=20,
+        use_ssl=True,
+        verify_certs=False,
+        sniff_on_start=False,
+        # refresh nodes after a node fails to respond
+        sniff_on_connection_fail=True,
+        # and also every 60 seconds
+        sniffer_timeout=60,
+        sniff_timeout=10,
+    )
+
+
+async def consume_logs(nw, inferenced_logs_queue):
+    async def subscribe_handler(msg):
+        data = msg.data.decode()
+        await inferenced_logs_queue.put(pd.read_json(data, dtype={"_id": object, "cluster_id": str, "ingest_at": str}))
+
+    await nw.subscribe(
+        nats_subject="inferenced_logs",
+        nats_queue="workers",
+        payload_queue=inferenced_logs_queue,
+        subscribe_handler=subscribe_handler,
+    )
+
+
+async def receive_logs(queue):
+    es = await setup_es_connection()
+    while True:
+        df = await queue.get()
+        await update_logs(es, df)
+
+async def update_logs(es, df):
+    # This function will be updating Opensearch logs which were inferred on by the DRAIN model.
+    model_keywords_dict = {"drain": {"Normal":  ["_id", "_op_type", "_index", "masked_log", "drain_pretrained_template_matched", "inference_model"],
```

Comment:
we should update anomaly level for Normal logs too.

Reply:
the default anomaly level is set to Normal though by the ingest plugin so we don't need to update that again right?

Reply:
https://github.com/tybalex/opni-ingest-plugin/pull/9
let's change the default value to ""